### PR TITLE
Feature/version

### PR DESCRIPTION
**Pull request by lpascal-ledger**
Adding firmware version checks using `semver` + corner cases (when using versions such as `2.1` which are expected from `speculos`).
This feature will help managing UX behavior changing depending on the SDK version.

### CODE REVIEW

File: tests/backend/test_interface.py [hunk 1]
```diff
@@ -46,11 +47,14 @@ def receive(self):
     def send_raw(self, *args, **kwargs):
         self.mock.send_raw(*args, **kwargs)
 
+    def finger_touch(self, *args, **kwargs):
```
Comment (agrojean-ledger) [inline tests/backend/test_interface.py:50]: Why add finger touch in this PR ? @lpascal-ledger

Comment (lpascal-ledger) [reply tests/backend/test_interface.py:50]: Glitching

File: src/ragger/firmware/versions.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        if major is not None and minor is None:
+            return max(v.value for v in cls if v.value.major == major)
+        if major is not None and minor is not None and patch is None:
+            return max(v.value for v in cls if (v.value.major == major and v.value.minor == minor))
+        try:
+            return max(
+                v.value for v in cls
+                if (v.value.major == major and v.value.minor == minor and v.value.patch == patch))
+        except ValueError:
+            raise ValueError(f"No version {major}.{minor}.{patch} found")
```
Comment (xchapron-ledger) [inline src/ragger/firmware/versions.py:46]: Why catching here and not above? Maybe you could let the default message?

**Review by xchapron-ledger — COMMENTED** [review]
You are fixing some `flake8` issue. Can you split them in an separate commit?
Also can you enforce the CI flake8 check? It's currently disbaled by `|| echo 1`?
If you do change the `.github/workflows/fast-checks.yml` can you fix the triggers? In my opinion, it would be better to have:
```
on:
  push:
    branches:
      - develop
      - master
  pull_request:
```
so that we don't have double tests on classical PR.

**Review by xchapron-ledger — APPROVED** [review]

File: src/ragger/firmware/versions.py (hunk 2 above)
Comment (lpascal-ledger) [reply src/ragger/firmware/versions.py:46]: Without management, the raised error would be `ValueError: max() arg is an empty sequence` which is not very helpfull.
But you're right this should be crafted for this case, but also with major only, or major + minor.
